core/transactions/staged_mutation.cxx

```diff
@@ -25,6 +25,23 @@
 namespace couchbase::core::transactions
 {
 
+void
+unstaging_state::wait_until_unstage_possible()
+{
+    std::unique_lock lock(mutex_);
+    cv_.wait(lock, [this]{ return in_flight_count_ < MAX_PARALLELISM; });
```

Comment:
If something goes wrong and the predicate never succeeds, do we wait here forever?

Reply:
At the moment yes, so this is something I should fix. Theoretically, since the mutation operations have timeouts this should not happen, but since we are using concurrency I can't be 100% certain that things can't go wrong so I'll probably add a timeout to `wait` - but I'm not sure what timeout I should use

Reply:
How about what's left of the transaction timeout?